### PR TITLE
Always hidden, hide '|'

### PR DESCRIPTION
**Pull request by s-valent**
#### What's this PR do?

- [x] Minor changes of interface
- [x] Allow to hide '|' button to have more space in status bar
- [x] Optional 'always hidden' section

#### What are the relevant Git tickets?

https://github.com/dwarvesf/hidden/issues/73
https://github.com/dwarvesf/hidden/issues/80

### CODE REVIEW

**Comment by phucledien** [comment]
Cool 👍 , thanks for the pr. I will take a look at it 😄

**Comment by phucledien** [comment]
I just took a quick look at it.

I believe that using right-click to open the context menu is a kind of anti-pattern and I'm not really sure if Apple allows it or not.

In the past version, we did use this and got reject. Anyway, just keep what you are doing here and I will give it a try. If Apple rejects it I will remove the right click part.

Another quick note: You should consider using `if let` or `guard let` to safely unwrap the optional 👍

File: README.md [hunk 1]
```diff
@@ -1,71 +1,16 @@
 <p align="center">
```
Comment (phucledien) [inline README.md:1]: Lol, please keep the original version of this file 😄

File: hidden/Features/StatusBar/StatusBarController.swift [hunk 2]
```diff
@@ -46,19 +75,38 @@ class StatusBarController {
...
             button.target = self
-            button.action = #selector(expandCollapseIfNeeded(_:))
+            
+            button.action = #selector(self.statusBarButtonClicked(sender:))
+            button.sendAction(on: [.leftMouseUp, .rightMouseUp])
+        }
+    }
+    
+    @objc func statusBarButtonClicked(sender: NSStatusBarButton) {
+        if NSApp.currentEvent!.type == NSEvent.EventType.leftMouseUp {
```
Comment (phucledien) [inline hidden/Features/StatusBar/StatusBarController.swift:106]: you should use optional here instead of force unwrap it 👍

**Comment by s-valent** [comment]
Thank you for your quick answer:)

I have some questions about right-click. It's important to leave some kind of other way of interacting with '<' to show the menu (not left-click) so '|' could be hidden and shown. Is everything apart from left-click is rejected? (may be option+click or something else is ok?)

**Comment by phucledien** [comment]
hi @s-valent , I have used it, and everything seems to work perfectly. But I think we should keep showing the menu context when a user clicks to the separator menu button so that the old user will not be surprised with the changes  🤣 

Would you mind to change it ❤️

File: hidden/Features/StatusBar/StatusBarController.swift [hunk 3]
```diff
@@ -46,19 +75,44 @@ class StatusBarController {
...
     }
     
     private func setupUI() {
+        self.setupContextMenu()
+        
         if let button = separateStatusBar.button {
             button.image = self.imgIconLine
         }
         
-        separateStatusBar.menu = self.getContextMenu()
```
Comment (phucledien) [inline hidden/Features/StatusBar/StatusBarController.swift:56]: You can keep it or add a popup menu action for the separator button 👍

**Comment by phucledien** [comment]
Hi @s-valent, can you update this pr 😢 

If you got any problems, just feel free to chat with me. I'm willing to help you 😄

**Comment by s-valent** [comment]
I'm really sorry for waiting so long(
I was busy last month passing exams, will be here soon fixing

**Comment by phucledien** [comment]
haha, @s-valent 😄  I really understand it.

please don't be worry, just update it whenever you feel comfortable.

**Pull request closed without merging**